src/com/aca/vardanmanasaryan/homework1/management/Spygadgets.java

```diff
@@ -0,0 +1,31 @@
+package com.aca.vardanmanasaryan.homework1.management;
+
+public abstract class Spygadgets implements SaleItem {
+public int price;
+public int quanity;
+protected final String name;
+protected final    int recordinghours;
+protected final String typeofrecording;
```

Comment:
putting here string you allow set here any value for the string which will cause problems. For instance, the client will be available  to pass here "DRUGS" value, which is not a good thing )